logstash-core/lib/logstash/shutdown_controller.rb

```diff
@@ -78,7 +82,7 @@ def shutdown_stalled?
       return false unless @reports.size == @report_every #
       # is stalled if inflight count is either constant or increasing
       stalled_event_count = @reports.each_cons(2).all? do |prev_report, next_report|
-        prev_report.inflight_count["total"] <= next_report.inflight_count["total"]
+        prev_report.inflight_count <= next_report.inflight_count
```

Comment:
@ph we have thought about this, the problem is that what we're trying to gauge here (is it stalled or not) is not a property between 2 reports, but of an array of reports. for this we test

for an array is condition X true?
if so, for the array, is condition Y true?

This means we need 2 Comparable strategies